src/security/context.cpp

```diff
@@ -332,6 +226,14 @@ class PolTaskCtx {
     ngx_log_debug(NGX_LOG_DEBUG_HTTP, req_log(), 0,
                   "refcount before decrement upon task %p completion: %d", this,
                   count);
+
+    // create a stack copy to safely call destructor later
+    // the task is allocated on the request pool, so by the end of this method
+    // maybe *this is no longer valid
+    alignas(Self) char self_copy_storage[sizeof(Self)];
+    std::memcpy(self_copy_storage, this, sizeof(Self));
+    Self *self_copy = reinterpret_cast<Self *>(self_copy_storage);
```

Comment:
Are you suggesting that on request pool destruction, the destructor of this class isn't called?

This is quite sketchy, wouldn't it be better to have the request pool hold a pointer so you can just use `this` to invoke the destructor?